runtime/include/cloe/databroker.hpp

```diff
@@ -0,0 +1,1254 @@
+    void set_value(const TValue* value)
+    {
+        // DRY
+        *this = *value;
+    }
+
+    const TValue* value() const
+    {
+        return &value_;
+    }
+    TValue* value()
+    {
+        // DRY
+        return const_cast<TValue*>(const_cast<const Container*>(this)->value());
+    }
+
+friend class Signal;
+
+}; // class Container
+
+/**
+ * @brief Signal-Descriptor
+ * @note: Design-Goals:
+ * - Design-#1: The class shall expose a uniform interface (via type-erasure)
+ * - Design-#2: The class shall be constructible via external helpers
+ * - Design-#3: CppCoreGuidelines CP-1 does not apply. See also: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconc-multi
+ * - Design-#4: Templates shall be instantiated explicitly
+ * @note: Implementation-Notes:
+ * - Implementation-#1: Type specific aspects are implemented in templates (Design-#1)
+ * - Implementation-#2: Objects are created via a factory-method (to prevent instances of incomplete initialization)
+ * - Implementation-#3: An access-token is used to have a public c'tor which in fact shall be inaccesible (Design-#2).
+ */
+class Signal
+{
+private:
+    /**
+     * Access-token for regulating API access (public -> private)
+     */
+    struct access_token{
+        explicit access_token(int){};
+    };
+    // Signal types
+    using signal_ptr_t = std::shared_ptr<Signal>;
+
+    // Getter-function types
+    template <typename TValue>
+    using typed_get_value_function_t = std::function<const TValue*()>;
+    using type_erased_get_value_function_t = std::any;
+    // Setter-function types
+    template <typename TValue>
+    using typed_set_value_function_t = std::function<void(const TValue*)>;
+    using type_erased_set_value_function_t = std::any;
+    // Event types
+    template <typename TValue>
+    using typed_value_changed_event_t = Event<const TValue*>;
+    using type_erased_value_changed_event_t = std::any;    
+    // Event trigger-function types
+    template<typename TValue>
+    using typed_on_value_change_event_function_t = std::function<void(const TValue*)>;
+    using type_erased_on_value_change_event_function_t = std::any;
+
+    /**
+     * Name of the signal
+     */
+    std::vector<std::string> names_{};
+    /**
+     * std::type_info of the signal
+     */
+    const std::type_info* type_ {nullptr};
+    /**
+     * getter-function
+     */
+    type_erased_get_value_function_t get_value_{};
+    /**
+     * setter-function
+     */
+    type_erased_set_value_function_t set_value_{};
+    /**
+     * Event which gets raised when the signal value changes
+     */
+    type_erased_value_changed_event_t value_changed_event_{};
+    /**
+     * Triggers the value-changed event
+     */
+    type_erased_on_value_change_event_function_t on_value_changed_{};
+    /**
+     * std::function returning the count of event subscribers
+     */
+    std::function<std::size_t()> subscriber_count_{};
+
+    /**
+     * @brief Private default c'tor

```

Comment:
In this project, we use `\` instead of `@`. I know it doesn't affect the semantics, but I'd like to be consistent.

Reply:
Done.